src/hotspot/share/jvmci/jvmciRuntime.cpp

```diff
@@ -844,6 +844,24 @@ void JVMCINMethodData::invalidate_nmethod_mirror(nmethod* nm) {
   }
 }
 
+void JVMCINMethodData::clear_nmethod_mirror_index() {
+  _nmethod_mirror_index = -1;
+}
+
+void JVMCINMethodData::relocate_nmethod_mirror(nmethod* nm) {
+  oop nmethod_mirror = get_nmethod_mirror(nm, /* phantom_ref */ false);
+  if (nmethod_mirror == nullptr) {
+    return;
+  }
+
+  JVMCIEnv* jvmciEnv = nullptr;
+  HotSpotJVMCI::InstalledCode::set_address(jvmciEnv, nmethod_mirror, (jlong)(nm));
```

Comment:
What's the sync story here? Any lock protecting this? If not, I wonder if readers are okay with inconsistencies. I haven't checked.